doc/source/user_guide.md

```diff
@@ -421,12 +422,59 @@ The number of dimensions is the number of continuous variables plus the number o
+
+   You'll encounter this error when the call is taking too long to complete on the server.
+   Most of the time, this will be during job execution or dataset upload/download.
+   I'll encourage you to read up on the [`handling timeouts`](#handling-timeouts) section to deal with these kind of errors.
+
+2. Validation errors
+
+   Validation errors happen due to bad user input. Our error message rely heavily on [HTTP status codes](https://en.wikipedia.org/wiki/List_of_HTTP_status_codes). In short, codes in the 400-499 range are user errors, and 500-599 are server errors. More on those later.
+
+   Here we'll cover the user errors, than you can remedy by modifying your parameters and trying again.
+   The syntax of the error message will always be of the following form:
+
+   ```text
+   Got error in HTTP request: POST https://company.octopize.app/reports. Error status 400 - privacy_metrics job status is not success: JobStatus.FAILURE
+   ```
+
+   You'll have: - the HTTP request method (`POST`, `GET`, etc...) - the endpoint that was affected (`/reports`) - the status (`400`) - an informational message that details the exact error that is happening (`privacy_metrics job status is not success: JobStatus.FAILURE`)
+
+   In this particular case, the user is calling the `/reports` endpoint, trying to generate a report. Generating a report needs a privacy metrics job to be successful to be able to show the metrics. However, in this case, the privacy job was in the `JobStatus.FAILURE` state.
+   The fix is then to go look at the error message that the privacy job threw up, launch another privacy job that is successful, and launch the generation of the report with the new privacy job once it is successful.
+
+3. `JobStatus.FAILURE`
+
+   Jobs that fail do not throw an exception. Rather, you have to inspect the `JobStatus` that is in the `status` property.
+
+   ```python
+   job = client.jobs.get_avatarization_job(job_id)
+   print(job.status) # JobStatus.FAILURE
+   ```
+
+   If the status is `JobStatus.FAILURE`, the `error_message` property will contain an explanation of the error.
+   You'll have to relaunch the job again with the appropriate modifications to your input.
+
+4. Internal error
+
+   Internal errors happen when there is an error on the server, meaning that we did not handle the error on our side, and something unexpected happened, for which we cannot give you an exact error message.
+   These come with a 500 HTTP status code, and the message is `internal error`.
+   In these cases, there is not much you can do except trying again with different parameters, hoping to not trigger the error again.
+
+   When these happen, our error monitoring software catches these and notifies us instantly. We will reach out to you to you to help you circumvent the issue if it's possible, while we investigate on our side. We'll be hard at work trying to resolve the bug, and push out a new version with the fix.
```

Comment:
Will we always reach out to them ? We don't exactly know who they are with sentry for instance...
```suggestion
   When these happen, our error monitoring software catches these and notifies us instantly. We will reach out to you to help you circumvent the issue if it's possible, while we investigate on our side. We'll be hard at work trying to resolve the bug, and push out a new version with the fix.
```

Reply:
The goal is to reach out to them, yes.